File: week8/이지훈/타겟 넘버.java
public class 타겟 넘버 {
    private static int result;

    public int solution(int[] numbers, int target) {
        dfs(0, target, numbers);

        return result;
    }

    private void dfs(int depth, int target, int[] numbers) {
        if (depth == numbers.length) {
            if (isTarget(numbers, target)) {
                result++;
            }

            return;
        }

        dfs(depth + 1, target, numbers);

        numbers[depth] *= -1;
        dfs(depth + 1, target, numbers);
        numbers[depth] *= -1;
    }

    private static boolean isTarget(int[] numbers, int target) {
        int sum = 0;

        for (int num : numbers) {
            sum += num;
        }

        if (sum == target) return true;

        return false;
    }

}

File: week8/윤여준/타겟 넘버.cpp
#include <bits/stdc++.h>
using namespace std;

int solve(int idx, int sum, vector<int>& nums, int target) {
    if(idx == nums.size()) {
        if(sum == target) return 1;
        return 0;
    }
    
    return solve(idx+1, sum + nums[idx], nums, target) + solve(idx+1, sum - nums[idx], nums, target);
}

int solution(vector<int> numbers, int target) {
    return solve(0, 0, numbers, target);
}